autoload/gitgutter/hunk.vim

```diff
@@ -300,22 +300,47 @@ function! s:preview(hunk_diff)
   let body_length = len(body)
   let previewheight = min([body_length, &previewheight])
 
-  silent! wincmd P
-  if !&previewwindow
-    noautocmd execute g:gitgutter_preview_win_location previewheight 'new'
-    set previewwindow
+  if has('nvim') && exists('*nvim_open_win')
+    let buf = nvim_create_buf(v:false, v:false)
+    let width = max(map(copy(lines), 'strdisplaywidth(v:val)'))
+    let height = len(lines)
+
+    let s:floating_win = nvim_open_win(
+          \ buf, v:true, {
```

Comment:
The `v:true` doesn't seem to work for me, i.e. my cursor doesn't enter the floating window.

Reply:
It does, and that's how `setlocal` works in the floating window. `noautocmd wincmd p` gets you back to the previous window. Checkout `:h api-floatwin`

Reply:
My cursor does not enter the floating window.  When I invoke a hunk preview the cursor stays where it is and the floating window is displayed below it.  If I move the cursor the floating window disappears (as expected when the cursor is in the original window).

Reply:
That's due to moving out of it later: https://github.com/aymanbagabas/vim-gitgutter/blob/3c75c97c3dfa12f0e868b13ed9344071b54a1d0a/autoload/gitgutter/hunk.vim#L336

Reply:
btw: it would be better to not use "noautocmd" there, for plugins tracking the current window etc.  At least nvim will trigger the autocmds for the floating window, but not when going back then.

Reply:
> That's due to moving out of it later

Ah, of course.  Thanks.

Reply:
> it would be better to not use "noautocmd" there

That's been there since the feature was introduced.  I can't remember now why I didn't want autocmds.